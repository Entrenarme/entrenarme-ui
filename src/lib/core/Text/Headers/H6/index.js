// @flow
import * as React from 'react';
import styled from 'styled-components';

import * as fonts from '../../../../helpers/fonts';
import colors from '../../../../helpers/colors';

const normalDesktop = 0.875;

const normalMobile = 0.875;

const sizeModifier = 0.25;

const textSizeDesktopTransform = props => {
  if (props.l) {
    return normalDesktop + sizeModifier;
  } else if (props.s) {
    return normalDesktop - sizeModifier;
  } else {
    return normalDesktop;
  }
};

const textSizeMobileTransform = props => {
  if (props.l) {
    return normalMobile + sizeModifier;
  } else if (props.s) {
    return normalMobile - sizeModifier;
  } else {
    return normalMobile;
  }
};

const marginBottomSize = props => {
  if (props === 'l') {
    return 1.875;
  } else if (props === 's') {
    return 0.625;
  } else {
    return 1.25;
  }
};

const StyledH6 = styled.h6`
  text-transform: uppercase;
  font-size: ${props => `${textSizeDesktopTransform(props.options)}rem`};
  font-weight: 400;
  font-family: ${fonts.mainFont};
  text-align: ${props => props.options.textAlign};
  color: ${props => colors[props.options.color].default};
  margin-top: 0rem;
  margin-left: 0rem;
  margin-right: 0rem;
  margin-bottom: ${props =>
    `${marginBottomSize(props.options.marginBottom)}rem`};

  /* Smartphones (portrait) ----------- */
  @media only screen and (max-width: 320px) {
    font-size: ${props => `${textSizeMobileTransform(props.options)}rem`};
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (min-width: 320px) and (max-width: 767px) {
    font-size: ${props => `${textSizeMobileTransform(props.options)}rem`};
  }

  /* iPads (portrait and landscape) ----------- */
  @media only screen and (min-width: 768px) and (max-width: 1223px) {
    /* Styles */
  }

  /* Desktops and laptops ----------- */
  @media only screen and (min-width: 1224px) {
    /* Styles */
  }

  /* Large screens ----------- */
  @media only screen and (min-width: 1824px) {
    /* Styles */
  }

  /* ----------- Retina Screens ----------- */
  @media screen and (min-device-width: 1200px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
    /* Styles */
  }
`;

type Props = {
  children: React.Node,
  color?: color,
  l?: boolean,
  s?: boolean,
  marginBottom?: 's' | 'm' | 'l',
  textAlign?: 'left' | 'center' | 'right',
};

const H6 = ({
  children,
  color,
  l,
  s,
  marginBottom,
  textAlign,
  ...rest
}: Props) => (
  <StyledH6 {...rest} options={{ color, l, s, marginBottom, textAlign }}>
    {children}
  </StyledH6>
);

H6.defaultProps = {
  color: 'primary',
  l: false,
  s: false,
  marginBottom: 'm',
  textAlign: 'left',
};

export default H6;
