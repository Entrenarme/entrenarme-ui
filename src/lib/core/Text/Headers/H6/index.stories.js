// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import H6 from './index';
import P from '../../Paragraph/index';

const ElementDecoration = storyFn => (
  <div style={{ padding: '3rem' }}>{storyFn()}</div>
);

storiesOf('Text/Headers/H6', module)
  .addDecorator(ElementDecoration)
  .add(
    'default',
    withInfo('Heading 6')(() => (
      <div>
        <H6>Heading 6</H6>
        <P>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
          fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
          justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
          nec auctor eros sagittis eget. Phasellus eleifend risus et arcu
          laoreet, vitae lobortis turpis feugiat. Maecenas pulvinar ultricies
          risus nec tincidunt. Donec tristique imperdiet dui at egestas.
          Praesent ornare et ante sit amet condimentum.
        </P>
      </div>
    )),
  );

storiesOf('Text/Headers/H6', module)
  .addDecorator(ElementDecoration)
  .add(
    'small',
    withInfo('Heading 6')(() => (
      <div>
        <H6 size="small">Heading 6</H6>
        <P>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
          fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
          justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
          nec auctor eros sagittis eget. Phasellus eleifend risus et arcu
          laoreet, vitae lobortis turpis feugiat. Maecenas pulvinar ultricies
          risus nec tincidunt. Donec tristique imperdiet dui at egestas.
          Praesent ornare et ante sit amet condimentum.
        </P>
      </div>
    )),
  );

storiesOf('Text/Headers/H6', module)
  .addDecorator(ElementDecoration)
  .add(
    'large',
    withInfo('Heading 6')(() => (
      <div>
        <H6 size="large">Heading 6</H6>
        <P>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
          fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
          justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
          nec auctor eros sagittis eget. Phasellus eleifend risus et arcu
          laoreet, vitae lobortis turpis feugiat. Maecenas pulvinar ultricies
          risus nec tincidunt. Donec tristique imperdiet dui at egestas.
          Praesent ornare et ante sit amet condimentum.
        </P>
      </div>
    )),
  );

storiesOf('Text/Headers/H6', module)
  .addDecorator(ElementDecoration)
  .add(
    'color warning',
    withInfo('Heading 6')(() => (
      <div>
        <H6 color="warning">Heading 6</H6>
        <P>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
          fermentum magna eu velit semper hendrerit. Aenean aliquet pretium
          justo, ut rhoncus magna euismod et. Donec condimentum congue sapien,
          nec auctor eros sagittis eget. Phasellus eleifend risus et arcu
          laoreet, vitae lobortis turpis feugiat. Maecenas pulvinar ultricies
          risus nec tincidunt. Donec tristique imperdiet dui at egestas.
          Praesent ornare et ante sit amet condimentum.
        </P>
      </div>
    )),
  );
