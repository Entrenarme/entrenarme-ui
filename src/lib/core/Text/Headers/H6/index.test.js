import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import H6 from './index';

test('renders "Heading" with no props', () => {
  const { container } = render(<H6>Heading</H6>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with large size', () => {
  const { container } = render(<H6 size="large">Heading</H6>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with small size', () => {
  const { container } = render(<H6 size="small">Heading</H6>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with color prop', () => {
  const { container } = render(<H6 color="warning">Heading</H6>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with text align', () => {
  const { container } = render(<H6 textAlign="center">Heading</H6>);
  expect(container.firstChild).toMatchSnapshot();
});
