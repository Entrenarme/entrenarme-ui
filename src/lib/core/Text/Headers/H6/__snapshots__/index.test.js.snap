// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Heading" with color prop 1`] = `
<h6
  class="sc-bdVaJa fWEKdF"
>
  Heading
</h6>
`;

exports[`renders "Heading" with large size 1`] = `
<h6
  class="sc-bdVaJa ddvHEy"
>
  Heading
</h6>
`;

exports[`renders "Heading" with no props 1`] = `
<h6
  class="sc-bdVaJa gnIBdk"
>
  Heading
</h6>
`;

exports[`renders "Heading" with small size 1`] = `
<h6
  class="sc-bdVaJa iOEiHO"
>
  Heading
</h6>
`;

exports[`renders "Heading" with text align 1`] = `
<h6
  class="sc-bdVaJa lhnWVl"
>
  Heading
</h6>
`;
